common-files/utils/httputils.mjs

```diff
@@ -234,18 +236,70 @@ const responseLogger = (req, res, next) => {
   }
 };
 
+const isEndpointWhitelisted = req => {
+  let result = false;
+  req.app.get(ENDPOINTS_WHITELISTED).forEach(e => {
+    if (result === false && e) {
+      result = req.url.match(e) != null;
+    }
+  });
+
+  return result;
+};
+
+const HEALTH_CHECK_REGEX = new RegExp(/^\/healthcheck\/?$/);
+const authenticationHandler = (req, res, next) => {
+  if (
+    req.url.match(HEALTH_CHECK_REGEX) != null ||
+    isEndpointWhitelisted(req) ||
+    req.get('x-api-key') === process.env.AUTHENTICATION_KEY
+  ) {
+    return next();
+  }
+
+  logger.warn('Unauthorized access!');
```

Comment:
Shouldn't that be en error?

Reply:
I honestly don't know. I thought that reporting an error would raise a concern that was addressed smoothly by the stack.